packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -73,7 +73,7 @@ typedef WillPopCallback = Future<bool> Function();
 /// [Navigator.pages] so that it no longer includes the corresponding [Page].
 /// (Otherwise, the page will be interpreted as a new page to show when the
 /// [Navigator.pages] list is next updated.)
-typedef PopPageCallback = bool Function(Route<dynamic> route, dynamic result);
+typedef PopPageCallback = bool? Function(Route<dynamic> route, dynamic result);
```

Comment:
It seems like either this should be not be nullable, or the code below in line 4817 should be checking whether it is null rather than asserting that it's not null.

Given that we've had this check expecting it not to be null (it previously would have thrown if null was returned by this function), I'm inclined to believe that this should not be nullable. Did it break any tests or analysis making it non-nullable?

Reply:
In `cupertino/route_test.dart` there are several tests using `buildNavigator` that return null for `Navigator.onPopPage`, which is currently a non-nullable `PopPageCallback`. In these tests they do not return a bool, so I thought I'd make it nullable to support the tests.

Should I make `Navigator.onPopPage` nullable instead? I think that will work.

Reply:
I think that makes sense, but I'd feel better if @chunhtai or @goderbauer agree :)

Reply:
We should **not** make the return type of PopPageCallback nullable here. Instead we need to fix the tests to not return null. In the tests, null was probably chosen because the tests don't invoke the callback? Presumably, you could just return true or false from it since it doesn't seem to matter?

Reply:
I would change the tests to something like:

```dart
      buildNavigator(
        pages: myPages,
        onPopPage: (Route<dynamic> route, dynamic result) {
          assert(false); // the test shouldn't call this.
          return true;
        },
        transitionDelegate: detector,
      )
```

Reply:
SGTM! Thanks!